src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1316,6 +1318,111 @@ testUnsafeSendChan TestTransport{..} = do
 
   takeMVar clientDone
 
+testRegistryMonitoring :: TestTransport -> Assertion
+testRegistryMonitoring TestTransport{..} = do
+  localNode <- newLocalNode testTransport initRemoteTable
+  remoteNode <- newLocalNode testTransport initRemoteTable
+  return ()
+
+  -- Local process. Test if local process will be removed from
+  -- registry when it dies.
+  box <- newEmptyMVar
+  runProcess localNode $ do
+    pid <- spawnLocal $ do
+                expect
+    register "test" pid
+    tpid <- whereis "test"
+    if tpid == Just pid
+      then do _ <- monitor pid
+              send pid ()
+              ProcessMonitorNotification{} <- expect
+              tpid1 <- whereis "test"
+              liftIO $ putMVar box (Nothing == tpid1)
+      else liftIO $ putMVar box False
+  True <- takeMVar box     
+  return ()
+
+  -- Remote process. Test if remote process entry is removed
+  -- from registry when process dies.
+  remote1 <- testRemote remoteNode 
+  runProcess localNode $
+    let waitpoll = do
```

Comment:
I think for sanity I need to remove `monitor` in local case also.

According to our (and erlang) specification if `ProcessDead` notification arrives to node, than all instances of `Process` should be removed from internal structures (registry...). If I'm explicitly linked to remote process and that process is dead than it's guaranteed that I'll eventually get `ProcessDead` notification. In this test I want to show that remote process will be removed from the registry even if it's not explicitly linked/monitor it. So explicit monitor breaks the test.

What I've think that having such semantics we may want to monitor registry values somehow, and wait for such event. I think I'll need to check code once again, it's quite possible that we already have this option.